src/Symfony/Bridge/Twig/Extension/AssetExtension.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Asset\Packages;
+
+/**
+ * Twig extension for the Symfony Asset component.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AssetExtension extends \Twig_Extension
+{
+    private $packages;
+
+    public function __construct(Packages $packages)
+    {
+        $this->packages = $packages;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('asset_path', array($this, 'getAssetPath')),
```

Comment:
I still think that we sould keep the old name, it's the best one. And it's a pitty that we have to struggle to find a new name just for BC reasons. I would prefer to have a ""harder"" approach like the one had before (only one layer at a time possible) and keep the same name.

Besides, keeping the old name makes it easier to upgrade, as the removed "absolute" and "version" params were mostly unused, and the asset name and package name remain in the new function, so there is almost full BC in the twig files if you choose to use the new layer, no need to rename anything.